core/src/main/java/org/elasticsearch/percolator/MultiDocumentPercolatorIndex.java

```diff
@@ -71,7 +72,16 @@ public void prepare(PercolateContext context, ParsedDocument parsedDocument) {
         try {
             MultiReader mReader = new MultiReader(memoryIndices, true);
             LeafReader slowReader = SlowCompositeReaderWrapper.wrap(mReader);
-            final IndexSearcher slowSearcher = new IndexSearcher(slowReader);
+            final IndexSearcher slowSearcher = new IndexSearcher(slowReader) {
+
+                @Override
+                public void search(Query query, Collector collector) throws IOException {
+                    BooleanQuery.Builder bq = new BooleanQuery.Builder();
+                    bq.add(query, BooleanClause.Occur.MUST);
+                    bq.add(Queries.newNestedFilter(), BooleanClause.Occur.MUST_NOT);
+                    super.search(bq.build(), collector);
```

Comment:
before we did this in a different place. The `QueryCollector` has a bunch of if statements that added a query if the document being percolated has a nested field. I removed that and add an extra clause here. I think this is a better place, because it is an implementation detail of the percolator index and the fact that a document has nested mappings shouldn't leak to other places in the code.